cmd/cosign/cli/dockerfile/resolve.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2021 The Sigstore Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dockerfile
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/sigstore/cosign/pkg/oci/remote"
+
+	"github.com/pkg/errors"
+)
+
+// ResolveDockerfileCommand rewrites the Dockerfile
+// base images to FROM <digest>.
+type ResolveDockerfileCommand struct {
+	Output string
+}
+
+// Exec runs the resolve dockerfile command
+func (c *ResolveDockerfileCommand) Exec(ctx context.Context, args []string) error {
+	if len(args) != 1 {
+		return flag.ErrHelp
+	}
+
+	dockerfile, err := os.Open(args[0])
+	if err != nil {
+		return fmt.Errorf("could not open Dockerfile: %w", err)
+	}
+	defer dockerfile.Close()
+
+	resolvedDockerfile, err := resolveDigest(dockerfile)
+	if err != nil {
+		return fmt.Errorf("failed extracting images from Dockerfile: %w", err)
+	}
+
+	if c.Output != "" {
```

Comment:
Should we handle these errors?

Reply:
forget about the errors, they can be fixed 🤩 thank you so much for your support, you are our hero, I still don't understand how do you manage all of your work, amazing effort 🤝👏!!!

Reply:
Done! Now, we also support `COPY --from=IMAGE` cases. 🎉 

cc: @dekkagaijin